pkg/broker/binding_operations.go

```diff
@@ -139,6 +139,18 @@ func userFromParams(bindingID string, password string, rawParams []byte) (*atlas
 		}
 	}
 
+	// The service_id and plan_id are required to be valid per the specification, despite
+	// not being used for bindings. We look them up to ensure they can be found in the catalog.
```

Comment:
My feeling is that this check belongs in the `Bind` function rather than in the `userFromParams` function, simply because it doesn't feel like a component of getting a user from a set of parameters.

Reply:
Very good point. @mrgunior can you move them up?

Reply:
Makes sense. Changes made.